### ISSUE TITLE
Sphinx build failure

### ISSUE THREAD

**Issue by renefritze** [readthedocs/sphinx-autoapi#263]
TL;DR: `sphinx.ext.viewcode` and autoapi conflict for me and also one should not point sphinx-autoapi to look into a directory structure where not every subdirectory is properly importable (?)

------------------------

I'm trying to switch our project from handwritten file walking to sphinx-autoapi, but I end up with a fatal error:
```
highlighting module code... [ 23%] pymor.analyticalproblems.instationary

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/usr/local/lib/python3.8/site-packages/sphinx/application.py", line 352, in build
    self.builder.build_update()
  File "/usr/local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 296, in build_update
    self.build(to_build,
  File "/usr/local/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 363, in build
    self.finish()
  File "/usr/local/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 614, in finish
    self.finish_tasks.add_task(self.gen_pages_from_extensions)
  File "/usr/local/lib/python3.8/site-packages/sphinx/util/parallel.py", line 49, in add_task
    res = task_func()
  File "/usr/local/lib/python3.8/site-packages/sphinx/builders/html/__init__.py", line 637, in gen_pages_from_extensions
    for pagename, context, template in pagelist:
  File "/usr/local/lib/python3.8/site-packages/sphinx/ext/viewcode.py", line 182, in collect_pages
    lines[start])
IndexError: list index out of range

Exception occurred:
  File "/usr/local/lib/python3.8/site-packages/sphinx/ext/viewcode.py", line 182, in collect_pages
    lines[start])
IndexError: list index out of range
The full traceback has been saved in /tmp/sphinx-err-ng90ol76.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

Copied the full trace from my local machine [sphinx-err-0a1r39l9.log](https://github.com/readthedocs/sphinx-autoapi/files/5881311/sphinx-err-0a1r39l9.log)

The entire log from our CI is [here](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/504395/raw)

There's an intermediate error before that:
```
reading sources... [ 89%] autoapi/wrapper/index
...
ModuleNotFoundError: No module named 'wrapper'
viewcode can't import wrapper, failed with error "No module named 'wrapper'"
```
Looks somewhat related and somewhat understandable since we do not have a top level module 'wrapper' to import, only `pymordemos/minimal_cpp_demo/wrapper.py`. Nevertheless sphinx-autoapi discovered that it seems?

If you're willing to use docker, this can be easily reproduced:
```
git clone https://github.com/pymor/pymor 
cd pymor
# with viewcode enabled, fails
git checkout f9d373dfdb72a89de39d9746c2cdf91eb29d0641
make docker_docs
```

While writing this issue I realized I should also test the build without the `viewcode` extension enabled.
This works OK (besides see warnings I'd need to eventually get rid of). 
If I then try to ignore the file in question by setting: 
```
autoapi_ignore = ['*/pymordemos/minimal_cpp_demo/wrapper.py']
```
I still get a (different) error. Looking into the dir I see that's not actually an importable module (no `__init__.py`). So I'm now ignoring the whole dir instead. Which finally results in a complete build.

Re-enabling `viewcode` will still fail though, so I think my actual problem is only related to that. I'll add a clean issue for that tomorrow.
